gtsam/hybrid/HybridGaussianConditional.cpp

```diff
@@ -192,11 +192,10 @@ bool HybridGaussianConditional::equals(const HybridFactor &lf,
 
   // Check the base and the factors:
   return BaseFactor::equals(*e, tol) &&
-         conditionals_.equals(e->conditionals_,
-                              [tol](const GaussianConditional::shared_ptr &f1,
-                                    const GaussianConditional::shared_ptr &f2) {
-                                return f1->equals(*(f2), tol);
-                              });
+         conditionals_.equals(
```

Comment:
Feels a little unnecessary. I generally prefer having type info directly in the code to make it more readable and easier to understand.

Reply:
I understand but it's a matter of taste. Will not undo.